File: ruby/day17_part1_2017.rb

steps = File.read('input.txt').to_i
buffer = [0]
current_position = 0

2017.times do |i|
  current_position = (current_position + steps) % buffer.length + 1
  buffer.insert(current_position, i + 1)
end

puts buffer[(current_position + 1) % buffer.length]

File: swift/day17_part1_2017.swift
import Foundation

extension Data {
    func toString() -> String {
        return String(data: self, encoding: .utf8) ?? ""
    }
}

let fileURL = URL(fileURLWithPath: "input.txt")
let steps = Int((try! String(contentsOf: fileURL)).trimmingCharacters(in: .whitespacesAndNewlines))!
var buffer = [0]
var currentPosition = 0

for i in 1...2017 {
    currentPosition = (currentPosition + steps) % buffer.count + 1
    buffer.insert(i, at: currentPosition)
}

print(buffer[(currentPosition + 1) % buffer.count])